### DISCUSSION TITLE
How can I set inventation link with my bot

### DISCUSSION THREAD

**Discussion by ghost** [python-telegram-bot/python-telegram-bot#2279] [Q&A]
Hello how can I set inventation link with my bot and how to track if the users are joining my chanel from that invait link.Thank I am searching for this for some time I still don`t find a solllution.

**Comment by Poolitzer** [comment]
hey, for a private supergroup/channel, every admin with add members right can obtain their own invitation link. To obtain one (initially or revoke), use [export_chat_invite_link](https://python-telegram-bot.readthedocs.io/en/stable/telegram.bot.html#telegram.Bot.export_chat_invite_link). After that you can use [get_chat](https://python-telegram-bot.readthedocs.io/en/stable/telegram.bot.html#telegram.Bot.get_chat) if you didnt save the link.

There is no direct way to track new members, but you can use [get_chat_member](https://python-telegram-bot.readthedocs.io/en/stable/telegram.bot.html#telegram.Bot.get_chat_member) to check if a specific user joined the chat.

**Reply by ghost** [reply]
I can't get the ivaite link This is the right way

```
    bot = context.bot
    chat_id = update.message.chat_id
    ivaite = bot.export_chat_invite_link(chat_id)
    print(ivaite)
```

**Reply by harshil21** [reply]
Use `chat_id = update.message.chat.id` there's no attribute `chat_id` in `Message`

**Reply by Bibo-Joshi** [reply]
[This](https://python-telegram-bot.readthedocs.io/en/stable/telegram.message.html#telegram.Message.chat_id) says otherwise @harshil21 ;)
Anyway @mikegrep you'll need to be more specific than "I can't get …". What's the problem? Do you have a traceback to show? See [here](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Ask-Right).

**Reply by harshil21** [reply]
> [This](https://python-telegram-bot.readthedocs.io/en/stable/telegram.message.html#telegram.Message.chat_id) says otherwise @harshil21 ;)

Ohh, my bad. Forgot that existed
